docs/source/running-jobs/environment-variables.rst

```diff
@@ -16,15 +16,25 @@ User-specified environment variables
 
 User-specified environment variables are useful for passing secrets and any arguments or configurations needed for your tasks. They are made available in ``file_mounts``, ``setup``, and ``run``.
 
-You can specify environment variables to be made available to a task in two ways:
+You can specify environment variables to be made available to a task in several ways:
 
-- ``envs`` field (dict) in a :ref:`task YAML <yaml-spec>`:
+- ``--env-file`` flag in ``sky launch/exec`` :ref:`CLI <cli>`, path to a `dotenv` file:
+
+  .. code-block:: text
+
+    # sky launch example.yaml --env-file my_app.env
+    # cat my_app.env
+    MYVAR=val
+    WANDB_API_KEY=MY_WANDB_API_KEY
+    HF_TOKEN=MY_HF_TOKEN
+
+- ``envs`` field (dict) in a :ref:`task YAML <yaml-spec>`: (takes precedence over the above)
 
   .. code-block:: yaml
 
     envs:
       MYVAR: val
-  
+
 - ``--env`` flag in ``sky launch/exec`` :ref:`CLI <cli>` (takes precedence over the above)
```

Comment:
By the way, can you double check in code whether this is the case, e.g., `sky launch --env ENV1 --env-file .env --env ENV2`. Does ordering matter?

Reply:
`--env` over `--env-file`

<img width="1447" alt="image" src="https://github.com/user-attachments/assets/96a14cfc-406f-4145-9389-a8e5e14a10ff">


And then the merged `env` will overrides the `envs` in `config.yaml`

<img width="680" alt="image" src="https://github.com/user-attachments/assets/6a3cfa9b-a55e-4c63-8f52-fefbd01b09c9">


The order doesn't matter

```bash
(sky) ➜  skypilot git:(dev/zeping/env_file_flag_to_doc) cat ~/hello-sky/env_test.yaml                                                                                                                   
# SkyPilot YAML configuration for a cluster with one node
workdir: .

resources:
  cloud: aws
  instance_type: t3.small

num_nodes: 1


envs:
  MYVAR: MYVAR_ENVS_IN_YAML
  WANDB_API_KEY: WANDB_API_KEY_ENVS_IN_YAML
  ENV_TEST: ENV_TEST_ENVS_IN_YAML
setup: |
  echo "Running setup."

run: |
  echo "MYVAR: $MYVAR"
  echo "WANDB_API_KEY: $WANDB_API_KEY"
  echo "ENV_TEST: $ENV_TEST"
(sky) ➜  skypilot git:(dev/zeping/env_file_flag_to_doc) cat ~/hello-sky/env_file.env                                                                                                                    
MYVAR=MYVAR_ENV_FILES
WANDB_API_KEY=WANDB_API_KEY_ENV_FILES
ENV_TEST=ENV_TEST_ENV_FILES
(sky) ➜  skypilot git:(dev/zeping/env_file_flag_to_doc) sky launch -c test_env --env MYVAR=MYVAR_CLI --env-file ~/hello-sky/env_file.env --env WANDB_API_KEY=WANDB_API_KEY_CLI ~/hello-sky/env_test.yaml

(task, pid=1928) MYVAR: MYVAR_CLI
(task, pid=1928) WANDB_API_KEY: WANDB_API_KEY_CLI
(task, pid=1928) ENV_TEST: ENV_TEST_ENV_FILES
```

Reply:
Changed the order in the doc @concretevitamin